### PR TITLE
Create jshint task for travis-ci & some ui changes

### PR DESCRIPTION
**Pull request by daveol**
Since we now have a easy way to test js files using grunt I think it should be a travis-ci task, there are also some ui changes in there

### CODE REVIEW

File: javascript/new-src/Gruntfile.js [hunk 1]
```diff
@@ -31,23 +28,23 @@ module.exports = function (grunt) {
...
-    }
+    jshintrc: true
    }
   },
   watch: {
    build: {
-    files: ['javascript/new-src/Gruntfile.js', 'javascript/new-src/src/js/app.js', 'javascript/new-src/src/index.html', 'javascript/new-src/src/css/style.css', 'javascript/new-src/src/views/*.html'],
-    tasks: ['jshint:dev', 'uglify:default', 'cssmin:default'],
+    files: ['javascript/new-src/Gruntfile.js',
+     'javascript/new-src/src/js/app.js', 'javascript/new-src/src/index.html',
```
Comment (peternewman) [inline javascript/new-src/Gruntfile.js]: Can you indent these to the [ and perhaps put one per line.

File: javascript/new-src/src/js/.jshintrc [hunk 2]
```diff
@@ -0,0 +1,29 @@
+{
```
Comment (peternewman) [inline javascript/new-src/src/js/.jshintrc]: If this is the same as the other one, is the system not clever enough to inherit? If not, can we have a symlink please?

File: javascript/new-src/src/js/app.js [hunk 3]
```diff
@@ -14,520 +14,582 @@
...
+ .factory('$ola', ['$http', '$window', 'OLA',
+  function ($http, $window, OLA) {
+   'use strict';
+   // once olad supports json post data postEncode can go away
+   // and the header in post requests too.
+   var postEncode = function (data) {
+    var PostData = [];
+    for (var key in data) {
+     if (key === 'd') {
+      PostData.push(key + '=' + data[key]);
```
Comment (peternewman) [inline javascript/new-src/src/js/app.js]: Can we just stick a comment in explaining why this doesn't get encoded please.

File: javascript/new-src/src/js/app.js [hunk 4]
```diff
@@ -14,520 +14,582 @@
...
-   $scope.Class = "";
+   $scope.Class = '';
+   $scope.Data = {
+    id: 0,
+    name: '',
+    add_ports: ''
+   };
+   $scope.$watchCollection('Data', function () {
+    if ($scope.Universes.indexOf($scope.Data.id) !== -1 ||
+     $scope.Data.name === '' ||
```
Comment (peternewman) [inline javascript/new-src/src/js/app.js]: Can these be indented to match the (

File: javascript/new-src/src/js/app.js [hunk 5]
```diff
@@ -14,520 +14,582 @@
...
+     $scope.Data.add_ports === '') {
+     $scope.buttonState = true;
+     return;
+    }
+    if ($scope.Data.id === null ||
+     $scope.Data.name === null ||
```
Comment (peternewman) [inline javascript/new-src/src/js/app.js]: Again.

File: javascript/new-src/src/js/app.js [hunk 6]
```diff
@@ -14,520 +14,582 @@
...
+     $scope.Data.add_ports === null) {
+     $scope.buttonState = true;
+     return;
+    }
+    if (typeof ($scope.Data.id) !== 'number' ||
+     typeof ($scope.Data.name) !== 'string' ||
```
Comment (peternewman) [inline javascript/new-src/src/js/app.js]: Again

File: javascript/new-src/src/js/app.js [hunk 7]
```diff
@@ -14,520 +14,582 @@
...
+     typeof ($scope.Data.add_ports) !==
+     'string') {
```
Comment (peternewman) [inline javascript/new-src/src/js/app.js]: Bring this onto the previous line.

File: javascript/new-src/src/js/app.js [hunk 8]
```diff
@@ -14,520 +14,582 @@
...
+     $scope.buttonState = true;
+     return;
+    }
+    $scope.buttonState = false;
+   });
+   $ola.get.ItemList().then(function (data) {
+    for (var u in data.universes) {
+     if ($scope.Data.id ===
+      parseInt(data.universes[u].id, 10)) {
```
Comment (peternewman) [inline javascript/new-src/src/js/app.js]: Again.

File: javascript/new-src/src/js/app.js [hunk 9]
```diff
@@ -14,520 +14,582 @@
...
+   $scope.ceil = function (i) {
+    return $window.Math.ceil(i);
    };
+   $scope.change = function () {
+    $ola.post.Dmx($scope.Universe, $scope.get);
+   };
+   $scope.page = function (d) {
+    if (d === 1) {
+     if (($scope.offset + 1) !==
+      $window.Math.ceil(OLA.MAX_CHANNEL_NUMBER / $scope.limit)) {
```
Comment (peternewman) [inline javascript/new-src/src/js/app.js]: Again.

File: javascript/new-src/Gruntfile.js [hunk 10]
```diff
@@ -31,23 +28,23 @@ module.exports = function (grunt) {
   jshint: {
    dev: ['javascript/new-src/Gruntfile.js', 'javascript/new-src/src/js/app.js'],
    options: {
-    globalstrict: true,
```
Comment (peternewman) [inline javascript/new-src/Gruntfile.js]: Have you lost this option entirely now?

**Comment by peternewman** [comment]
Looks good, just some minor comments.

File: javascript/new-src/src/js/.jshintrc (hunk 2 above)
Comment (daveol) [inline javascript/new-src/src/js/.jshintrc]: not the same on has globals for nodejs (Gruntfile.js) and one for angular and jquery (app.js) so they are different

File: javascript/new-src/src/js/app.js (hunk 9 above)
Comment (daveol) [inline javascript/new-src/src/js/app.js]: we will go over the 80 characters per line limit

File: javascript/new-src/Gruntfile.js (hunk 10 above)
Comment (daveol) [inline javascript/new-src/Gruntfile.js]: yes

File: javascript/new-src/src/js/.jshintrc (hunk 2 above)
Comment (peternewman) [inline javascript/new-src/src/js/.jshintrc]: How fussy is the config file, can you put all the options in one file and just ignore some of them?

File: javascript/new-src/src/js/app.js (hunk 9 above)
Comment (peternewman) [inline javascript/new-src/src/js/app.js]: Okay, you could also wrap the divide across two lines, or move everything after ceil( onto a further indented line.

File: javascript/new-src/src/js/.jshintrc (hunk 2 above)
Comment (daveol) [inline javascript/new-src/src/js/.jshintrc]: they don't merge

File: javascript/new-src/Gruntfile.js (hunk 10 above)
Comment (peternewman) [inline javascript/new-src/Gruntfile.js]: Okay

File: javascript/new-src/src/js/.jshintrc (hunk 2 above)
Comment (peternewman) [inline javascript/new-src/src/js/.jshintrc]: That's a shame.

**Comment by peternewman** [comment]
LGTM @nomis52 do you want to take a look?

**Comment by nomis52** [comment]
LGTM

**Pull request merged**